nextjs-interface/src/app/ui/dashboard/DataGraph.tsx

```diff
@@ -0,0 +1,621 @@
+"use client";
+
+import { Card, CardContent, CardTitle } from "@/components/ui/card";
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+} from "recharts";
+import React, { useContext, useEffect, useState, Suspense } from "react";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import useFindIpById, {
+  useAllEsp,
+  useFetchAllData,
+  useFetchData,
+} from "@/lib/data";
+import { differenceInDays, endOfWeek, format, startOfWeek } from "date-fns";
+import { DateRange } from "react-day-picker";
+import { DateRangeElement } from "@/app/ui/dashboard/DateRangeElement";
+import { useRouter, useSearchParams } from "next/navigation";
+import { Check, CornerLeftDown, CornerRightDown } from "lucide-react";
+import { ThemeContext } from "@/lib/Theme";
+import { avgData } from "@/lib/context";
+
+export default function DataGraph() {
+  const router = useRouter();
+
+  const [date, setDate] = useState<DateRange | undefined>(() => {
+    const now = new Date();
+    return {
+      from: startOfWeek(now),
+      to: endOfWeek(now),
+    };
+  });
+
+  const [selectedEsp, setSelectedEsp] = useState<string | undefined>(undefined);
+  const [selectedEsp2, setSelectedEsp2] = useState<string | undefined>(
+    undefined,
+  );
+  const [selectedOption, setSelectedOption] = useState<string>("1");
+
+  const dateRangeInDays =
+    date?.from && date?.to ? differenceInDays(date.to, date.from) : 0;
+
+  const ip1 = useFindIpById(selectedEsp ? selectedEsp : "");
```

Comment:
Are executed even if the corresponding graph is not displayed. Must be used only if we nee the data and have an id to query.